### ISSUE TITLE
`F821: Undefined name` in `match` capture patterns

### ISSUE THREAD

**Issue by mikeroll** [charliermarsh/ruff#3096]
First of all, big thanks and congrats on shipping `match` support! Looks like it's not quite perfect yet as I catch a false-positive F821 with a very simple snippet:
```python
def test(provided: int) -> int:
    match provided:
        case captured:
            return captured  # F821 Undefined name `captured`
```
```sh
❯ ruff --version
ruff 0.0.250
```

**Comment by charliermarsh** [comment]
Thank you!

**Comment by charliermarsh** [comment]
Will fix this. Please keep filing issues you run into with `match`. My focus was primarily on getting the syntax working (and traversals working within the linter), so the rules probably have rough edges.

**Comment by dpassen** [comment]
Not sure if this is the same issue or just very similar:

```python
from dataclasses import dataclass


@dataclass
class Car:
    make: str
    model: str


match Car("Toyota", "Corolla"):
    case Car("Toyota", model):
        print(f"Nice #{model}. I love what you do for me, Toyota!")

```

also yields an F821 Undefined name 'model'

**Comment by charliermarsh** [comment]
Same issue!
